owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt

```diff
@@ -278,14 +251,15 @@ abstract class DrawerActivity : ToolbarActivity() {
      * @param accountName The account name to be set
      */
     private fun accountClicked(accountName: String) {
-        if (AccountUtils.getCurrentOwnCloudAccount(applicationContext).name != accountName) {
-            AccountUtils.setCurrentOwnCloudAccount(
-                applicationContext,
-                accountName
-            )
-            // Refresh dependencies to be used in selected account
-            initDependencyInjection()
-            restart()
+        if (drawerViewModel.getCurrentAccount(this).name != accountName) {
+            if (drawerViewModel.setCurrentAccount(applicationContext, accountName)) {
```

Comment:
What does this if check exactly? You didn't need to check anything before

Reply:
It verifies that the `setCurrentAccount` was performed properly.